extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,417 @@
+      /** using Iframely placeholder image */
+      let src = this.constructPlaceholderSrc_();
+      src = addParamsToUrl(src, this.options_);
+      return createElementWithAttributes(
+        this.element.ownerDocument,
+        'amp-img',
+        {
+          'src': src,
+          'layout': 'fill',
+          'placeholder': '',
+        }
+      );
+    } else {
+      return null;
+    }
+  }
+
+  /** @override */
+  layoutCallback() {
+    /** attach iFrame */
+    const me = this;
+    this.iframe_ = this.element.ownerDocument.createElement('iframe');
+    setStyle(this.iframe_, 'border', '0');
+    this.iframe_.setAttribute(
+      'allow',
+      'encrypted-media *; accelerometer *; gyroscope *; picture-in-picture *; camera *; microphone *;'
+    );
+    this.iframe_.setAttribute(
+      'sandbox',
+      'allow-scripts allow-same-origin allow-popups allow-forms allow-presentation'
+    );
+    this.src_ = addParamsToUrl(this.src_, this.options_);
+    this.iframe_.src = this.src_;
+    this.applyFillContent(this.iframe_);
+    this.element.appendChild(this.iframe_);
+
+    /**
+     * Handling Iframely messages
+     * @param {window.event.source} event - event to check
+     * */
+    function receiveMessage(event) {
+      const iframes = me.element.getElementsByTagName('iframe');
+      if (findIframeByContentWindow(iframes, event.source)) {
+        me.handleEvent(me, event);
+      }
+    }
+    window.addEventListener('message', receiveMessage, false);
+    return this.loadPromise(this.iframe_);
+  }
+
+  /**
+   * Handles Iframely events: widget sizing, cancel, decorate cards
+   * @param {AmpIframely} me - instance of an active component
+   * @param {window.event} event - Iframely message with a method to apply
+   * */
+  handleEvent(me, event) {
+    let data = null;
+    try {
+      data = JSON.parse(event.data);
+    } catch (e) {
+      // Do nothing - likely not Iframely message if JSON errs
+    }
+    if (data) {
+      if (data.method === 'resize') {
+        /** Set the size of the card according to the message from Iframely */
+        const height = this.addBorderHeight_(me, data['height']);
+        me.attemptChangeHeight(height).catch(() => {});
+      }
+      if (data.method === 'setIframelyEmbedData') {
+        /** apply Iframely card styles if present */
+        const media = data['data']['media'] || null;
+        if (media && media['frame_style']) {
+          const styles = media['frame_style'].split(';');
+          styles.forEach(function(style) {
+            const props = style.split(':');
+            if (props.length === 2) {
+              const styleName = props[0].trim(),
+                styleValue = props[1].trim();
+              switch (styleName) {
+                case 'border':
+                  /** Because of the border-box sizing, changing border width doesn't change the size of the box. */
+                  setStyle(me.element, 'border', styleValue);
```

Comment:
You're could still be resizing if you set `border-box` dynamically.

Consider removing `border` changes: You can use a transparent background iframe. (`border-radius` and `box-shadow` is fine).

Reply:
Thanks. We'll be removing dynamic `border` changes by placing them into `data-border` attribute of the component.

Reply:
The border depends on the content itself (hence the dynamic change preference) and also on the user's settings. CSS is the last resort - and we'd need to add a few more descriptors to support it anyway. It will end up being way less user-friendly. 

An attribute should do fine.

Reply:
> An attribute should do fine.

No, since the extension loads dynamically, therefore the attribute is interpreted later and will cause Content Layout Shift.

Please implement this border inside the iframe, it should be possible if the iframe and its inner document have `background: transparent`.